File: frontend/src/types/sharedFile.ts
export interface SharedFile {
  id: string;
  fileName: string;
  originalFileName: string;
  contentType: string;
  fileSize: number;
  fileSizeFormatted: string;
  description?: string;
  tags: string[];
  uploaderId: string;
  uploaderName: string;
  groupId?: string;
  groupName?: string;
  shareType: 'group' | 'company' | 'department';
  status: 'active' | 'deleted' | 'archived';
  downloadCount: number;
  createdAt: string;
  updatedAt?: string;
  versions: FileVersion[];
  canEdit: boolean;
  canDelete: boolean;
  canDownload: boolean;
}

export interface FileVersion {
  id: string;
  versionNumber: number;
  fileName: string;
  fileSize: number;
  fileSizeFormatted: string;
  changeDescription?: string;
  uploadedById: string;
  uploadedByName: string;
  createdAt: string;
}

export interface UploadFileRequest {
  file: File;
  description?: string;
  tags: string;
  groupId?: string;
  shareType: 'group' | 'company' | 'department';
}

export interface UpdateFileRequest {
  description?: string;
  tags?: string;
  status?: 'active' | 'deleted' | 'archived';
}

export interface ShareFileRequest {
  fileId: string;
  shareType: 'employee' | 'group' | 'department' | 'company';
  employeeId?: string;
  groupId?: string;
  departmentId?: string;
  permission: 'view' | 'download' | 'edit';
  message?: string;
  expiresAt?: string;
}

export interface FileShare {
  id: string;
  fileId: string;
  fileName: string;
  sharedById: string;
  sharedByName: string;
  sharedWithId?: string;
  sharedWithName?: string;
  shareType: 'employee' | 'group' | 'department' | 'company';
  permission: 'view' | 'download' | 'edit';
  message?: string;
  expiresAt?: string;
  sharedAt: string;
  accessedAt?: string;
  status: 'active' | 'expired' | 'revoked';
}

export interface StorageUsage {
  usageBytes: number;
  usageFormatted: string;
}

File: backend/StaffManagementSystem/Models/SharedFile.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StaffManagementSystem.Models
{
    public class SharedFile
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required(ErrorMessage = "File name is required")]
        [StringLength(255)]
        public string FileName { get; set; } = string.Empty;

        [Required(ErrorMessage = "Original file name is required")]
        [StringLength(255)]
        public string OriginalFileName { get; set; } = string.Empty;

        [StringLength(100)]
        public string ContentType { get; set; } = string.Empty;

        public long FileSize { get; set; }

        [Required]
        [StringLength(500)]
        public string FilePath { get; set; } = string.Empty;

        [StringLength(64)]
        public string? FileHash { get; set; } // For duplicate detection

        [StringLength(500)]
        public string? Description { get; set; }

        [StringLength(200)]
        public string? Tags { get; set; } // JSON array of tags

        [Required]
        [ForeignKey("Uploader")]
        public string UploaderId { get; set; } = string.Empty;

        [ForeignKey("Group")]
        public string? GroupId { get; set; } // null means company-wide sharing

        [StringLength(20)]
        public string ShareType { get; set; } = "group"; // group, company, department

        [StringLength(20)]
        public string Status { get; set; } = "active"; // active, deleted, archived

        public int DownloadCount { get; set; } = 0;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public DateTime? DeletedAt { get; set; }

        // Navigation properties
        public User? Uploader { get; set; }
        public Group? Group { get; set; }
        public ICollection<FileShare> FileShares { get; set; } = new List<FileShare>();
        public ICollection<FileVersion> Versions { get; set; } = new List<FileVersion>();

        public SharedFile()
        {
            Id = Guid.NewGuid().ToString();
        }
    }
}